### DISCUSSION TITLE
How to apply template using Python

### DISCUSSION THREAD

**Discussion by nathanlukeIDEA** [idea-statica/ideastatica-public#381] [Help Q&A] [answered]
I have a  small question, I would like to apply a template file to the blank connection. Can I know how to set the template? I am doing this using Python.

_Originally posted by @umesh14398 in https://github.com/idea-statica/ideastatica-public/issues/367#issuecomment-1825064403_

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi @umesh14398, 

Something like this:

```python
import clr
import sys
import json
import math
import os

idea_path = "C:\Program Files\IDEA StatiCa\StatiCa 22.1"
sys.path.append(idea_path);

clr.AddReference('IdeaStatiCa.Plugin')
clr.AddReference("IdeaRS.OpenModel")

from IdeaStatiCa.Plugin import ConnHiddenClientFactory
from IdeaRS.OpenModel import OpenModel
from IdeaRS import OpenModel as IOM

factory = ConnHiddenClientFactory(idea_path)
con_client = factory.Create()

#Open the Project
client.OpenProject(project)

#Get the connectionId
projectInfo = client.GetProjectInfo()
connections = projectInfo.Connections
connection = connections[0].Identifier

#Apply template
client.ApplyTemplate(connection, 'baseplate_template_moment.contemp', IOM.Connection.ApplyConnTemplateSetting())

client.Save()
client.CloseProject()
client.Close()
```

The .contempt file can be created by exporting a template from an existing connection using the developer tab.
![image](https://github.com/idea-statica/ideastatica-public/assets/93578548/6e5c09e8-84bc-4ca2-bbf9-387db51ceab6)

**Reply by umesh14398** [reply]
@nathanlukeIDEA  thank you so much for the response. I have given the conTemplateSetting as null and tried to apply the template. It has applied.
